org.eclipse.lyo.oslc4j.provider.jena/src/main/java/org/eclipse/lyo/oslc4j/provider/jena/JenaModelHelper.java

```diff
@@ -851,14 +877,15 @@ else if (o.isResource())
 						}
 						else
 						{
-							final Object nestedBean = setMethodComponentParameterClass.newInstance();
-
+							//final Object nestedBean = setMethodComponentParameterClass.newInstance();
+                                                        final Object nestedBean = Modifier.isAbstract(setMethodComponentParameterClass.getModifiers()) ? getResourceClass(nestedResource, rdfTypeClasses).newInstance() : setMethodComponentParameterClass.newInstance();
```

Comment:
what does this line do?

Reply:
also, we might want to resolve the class even if it is not abstract, in order to instantiate the most concrete Java class available for the RDF resource hierarchy

Reply:
It checks whether the type of the property is abstract, in that case it searches a concrete type in the mapping definition. You're right, this logic should apply also for concrete types that were subclassed also and code should try to find the most concrete implementation. Missed that.